frontend/packages/console-dynamic-plugin-sdk/src/app/k8s/actions/k8s.ts

```diff
@@ -0,0 +1,281 @@
+import * as _ from 'lodash';
+import { ActionType as Action, action } from 'typesafe-actions';
+import { K8sModel, MatchLabels, Selector } from '../../../api/common-types';
+import { K8sResourceCommon, FilterValue } from '../../../extensions/console-types';
+import { getReferenceForModel } from '../../../utils/k8s/k8s-ref';
+import { k8sList, k8sGet } from '../../../utils/k8s/k8s-resource';
+import { k8sWatch } from '../../../utils/k8s/k8s-utils';
+import { getImpersonate } from '../../core/reducers/coreSelectors';
+
+type K8sResourceKind = K8sResourceCommon & {
+  spec?: {
+    selector?: Selector | MatchLabels;
+    [key: string]: any;
+  };
+  status?: { [key: string]: any };
+  data?: { [key: string]: any };
+};
+
+export enum ActionType {
+  StartWatchK8sObject = 'startWatchK8sObject',
+  StartWatchK8sList = 'startWatchK8sList',
+  ModifyObject = 'modifyObject',
+  StopWatchK8s = 'stopWatchK8s',
+
+  Errored = 'errored',
+  Loaded = 'loaded',
+  BulkAddToList = 'bulkAddToList',
+  UpdateListFromWS = 'updateListFromWS',
+  FilterList = 'filterList',
+}
+
+type K8sEvent = { type: 'ADDED' | 'DELETED' | 'MODIFIED'; object: K8sResourceKind };
+
+export const updateListFromWS = (id: string, k8sObjects: K8sEvent[]) =>
+  action(ActionType.UpdateListFromWS, { id, k8sObjects });
+export const loaded = (id: string, k8sObjects: K8sResourceKind | K8sResourceKind[]) =>
+  action(ActionType.Loaded, { id, k8sObjects });
+
+export const bulkAddToList = (id: string, k8sObjects: K8sResourceKind[]) =>
+  action(ActionType.BulkAddToList, { id, k8sObjects });
+
+export const startWatchK8sObject = (id: string) => action(ActionType.StartWatchK8sObject, { id });
+export const startWatchK8sList = (id: string, query: { [key: string]: string }) =>
+  action(ActionType.StartWatchK8sList, { id, query });
+export const modifyObject = (id: string, k8sObjects: K8sResourceKind) =>
+  action(ActionType.ModifyObject, { id, k8sObjects });
+export const stopWatchK8s = (id: string) => action(ActionType.StopWatchK8s, { id });
+
+export const errored = (id: string, k8sObjects: any) =>
+  action(ActionType.Errored, { id, k8sObjects });
+export const filterList = (id: string, name: string, value: FilterValue) =>
+  action(ActionType.FilterList, { id, name, value });
+
+const WS = {} as { [id: string]: WebSocket & any };
```

Comment:
I've removed the lint exceptions @invincibleJai had to add for the inconsistent return values for the functions below, however I have not yet addressed:
1. Avoiding use of these unscoped variables.
2. Removing some nested async functions below